src/hotspot/share/gc/z/zOopClosures.hpp

```diff
@@ -67,21 +67,21 @@ class ZPhantomIsAliveObjectClosure : public BoolObjectClosure {
 
 class ZPhantomKeepAliveOopClosure : public ZRootsIteratorClosure {
 public:
-  virtual void do_oop(oop* p) override;
-  virtual void do_oop(narrowOop* p) override;
+  virtual void do_oop(oop* p);
+  virtual void do_oop(narrowOop* p);
 
-  virtual ZNMethodEntry nmethod_entry() const override {
-    return ZNMethodEntry::VerifyDisarmed;
+  virtual ZNMethodEntry nmethod_entry() const {
+    return ZNMethodEntry::None;
```

Comment:
How about letting `ZRootsIteratorClosure::nmethod_entry()` be
```
ShouldNotReachHere();
return ZNMethodEntry::None;
```
instead of pure virtual, so that we can remove these unused overrides and catch errors in case a closure doesn't override this but we then apply it on nmethods? Alternatively add a `ShouldNotReachHere()` in these unused overrides?